src/storage/txn/actions/prewrite.rs

```diff
@@ -412,6 +437,78 @@ impl<'a> PrewriteMutation<'a> {
         .into())
     }
 
+    fn check_assertion<S: Snapshot>(
+        &self,
+        reader: &mut SnapshotReader<S>,
+        write: &Option<(Write, TimeStamp)>,
+        write_loaded: bool,
+    ) -> Result<(Option<(Write, TimeStamp)>, bool)> {
+        if self.assertion == Assertion::None
+            || self.txn_props.assertion_level == AssertionLevel::Off
+        {
+            return Ok((None, false));
+        }
+
+        let mut reloaded_write = None;
+        let mut reloaded = false;
+
+        // To pass the compiler's lifetime check.
+        let mut write = write;
+
+        if write_loaded
+            && write.as_ref().map_or(
+                false,
+                |(w, _)| matches!(w.gc_fence, Some(gc_fence_ts) if !gc_fence_ts.is_zero()),
+            )
+        {
+            // The previously-loaded write record has an invalid gc_fence. Regard it as none.
+            write = &None;
+        }
+
+        // Load the most recent version if prev write is not loaded yet, or the prev write is not
+        // a data version (`Put` or `Delete`)
+        let need_reload = !write_loaded
+            || write.as_ref().map_or(false, |(w, _)| {
+                w.write_type != WriteType::Put && w.write_type != WriteType::Delete
+            });
+        if need_reload {
+            let reload_ts = write.as_ref().map_or(TimeStamp::max(), |(_, ts)| *ts);
+            reloaded_write = reader.get_write_with_commit_ts(&self.key, reload_ts)?;
```

Comment:
How could we measure the reload or check cost quickly? We may need some utilities maybe some counter?

Reply:
Do  you mean add a metric here? I think that's a good idea